staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -466,13 +554,31 @@ func (t *tracker) Delete(gvr schema.GroupVersionResource, ns, name string) error
 		return errors.NewNotFound(gvr.GroupResource(), name)
 	}
 
-	delete(objs, namespacedName)
-	for _, w := range t.getWatches(gvr, ns) {
+	if objMeta, err := meta.Accessor(obj); err == nil {
+		if len(objMeta.GetFinalizers()) != 0 {
+			dt := metav1.Time{Time: t.clk.Now()}.Rfc3339Copy()
+			objMeta.SetDeletionTimestamp(&dt)
+			return t.doUpdateLocked(gvr, obj)
+		}
+	}
+
+	return t.doDeleteLocked(gvr, namespacedName)
+}
+
+func (t *tracker) doDeleteLocked(gvr schema.GroupVersionResource, name types.NamespacedName) error {
+	objs := t.objects[gvr]
+	obj := objs[name]
+	delete(objs, name)
+	for _, w := range t.getWatches(gvr, name.Namespace) {
 		w.Delete(obj.DeepCopyObject())
 	}
 	return nil
 }
 
+// AncientTime is a non-zero time far back in the past.
+// Can be used to set the deletion timestamp.
+var AncientTime = metav1.Date(1600, 1, 1, 0, 0, 0, 0, time.UTC)
```

Comment:
this is a weird place to add a variable like this... I wouldn't expect to need this

Reply:
Many existing tests uses zero time as the deletion timestamp. But that will get lost when going around json marshaling and unmashaling. I want to use this `AncientTime` as a safe replacement for zero time. I'm not sure if it is safe to just replace them with `metav1.Now()`.